src/opnsense/service/templates/OPNsense/IDS/suricata.yaml

```diff
@@ -254,121 +263,126 @@ outputs:
+            # verdict: yes
+{%      if not helpers.empty('OPNsense.IDS.general.eveLog.smtp.enable') %}
+        - smtp:
+            extended: {{ 'no' if helpers.empty('OPNsense.IDS.general.eveLog.smtp.extended') else 'yes' }} # enable this for extended logging information
             # this includes: bcc, message-id, subject, x_mailer, user-agent
             # custom fields logging from the list:
             #  reply-to, bcc, message-id, subject, x-mailer, user-agent, received,
             #  x-originating-ip, in-reply-to, references, importance, priority,
             #  sensitivity, organization, content-md5, date
+{%          if not helpers.empty('OPNsense.IDS.general.eveLog.smtp.custom') %}
+            custom: [{{ OPNsense.IDS.general.eveLog.smtp.custom }}]
+{%          else %}
             #custom: [received, x-mailer, x-originating-ip, relays, reply-to, bcc]
+{%          endif %}
             # output md5 of fields: body, subject
             # for the body you need to set app-layer.protocols.smtp.mime.body-md5
             # to yes
             #md5: [body, subject]
+{%      else %}
+        #- smtp
+{%      endif %}
 
-        #- dnp3
-        #- ftp
-        #- rdp
-        #- nfs
-        #- smb
-        #- tftp
-        #- ike
-        #- dcerpc
-        #- krb5
-        #- bittorrent-dht
-        #- snmp
-        #- rfb
-        #- sip
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.dnp3.enable') else '' }}- dnp3
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.ftp.enable') else '' }}- ftp
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.rdp.enable') else '' }}- rdp
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.nfs.enable') else '' }}- nfs
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.smb.enable') else '' }}- smb
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.tftp.enable') else '' }}- tftp
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.ike.enable') else '' }}- ike
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.dcerpc.enable') else '' }}- dcerpc
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.krb5.enable') else '' }}- krb5
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.bittorrentDht.enable') else '' }}- bittorrent-dht
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.snmp.enable') else '' }}- snmp
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.rfb.enable') else '' }}- rfb
+        {{ '#' if helpers.empty('OPNsense.IDS.general.eveLog.sip.enable') else '' }}- sip
+{%      if not helpers.empty('OPNsense.IDS.general.eveLog.quic.enable') %}
+        - quic:
+            # ja4 hashes in quic records will never be logged unless
+            # the following is set to on. (Default off)
+            ja4: {{ 'on' if helpers.empty('OPNsense.IDS.general.eveLog.tls.custom') or 'ja4' in OPNsense.IDS.general.eveLog.tls.custom else 'off' }}
```

Comment:
QUIC's JA4 support is enabled if JA4 is enabled for TLS. We should consider whether a standalone setting for QUIC is preferred.